library/std/src/lib.rs

```diff
@@ -351,6 +351,7 @@
 #![feature(slice_ptr_get)]
 #![feature(slice_range)]
 #![feature(std_internals)]
+#![feature(str_as_str)]
```

Comment:
Why is this needed? You didn't make any other changes in `std` -- did the new method break something?

Reply:
It's been used by a test for an error message specifically about lack of `str.as_str()` support. I've deleted that check.